### PR TITLE
chore: Clean up root directory

### PR DESCRIPTION
**Pull request by cpitclaudel**
- Remove unused vendored version of Coco (keep only frame files)
- Remove unused Vim and Emacs utils (now maintained out of tree)
- Move release instructions to `docs/dev`
- Move `interact.py` to `Scripts/`
- Move LaTeX style to `docs/`
- Remove `refman-release` Makefile target

### CODE REVIEW

File: Util/Emacs/README [hunk 1]
```diff
@@ -1,2 +0,0 @@
-Emacs support for dafny is provided by the boogie-friends package, available from MELPA.
```
Comment (MikaelMayer) [inline Util/Emacs/README:1]: Why did you delete this file? It seems like a good pointer. Is there another pointer to Emacs somewhere visible?

File: Makefile [hunk 2]
```diff
@@ -15,9 +15,6 @@ ${DIR}/boogie/Binaries/Boogie.exe:
 refman:
 	make -C ${DIR}/docs/DafnyRef
 
-refman-release:
```
Comment (MikaelMayer) [inline Makefile:18]: Why are you confident about deleting this?

**Review by MikaelMayer — CHANGES_REQUESTED** [review]
I like the idea of cleaning up the root directory, but I need some explanations.

File: Makefile [hunk 3]
```diff
@@ -15,9 +15,6 @@ ${DIR}/boogie/Binaries/Boogie.exe:
...
-	make -C ${DIR}/docs/DafnyRef release
-
```
[the remark this answers is not in the document]
Comment (cpitclaudel) [reply Makefile:20]: Correct

File: Util/Emacs/README (hunk 1 above)
Comment (cpitclaudel) [reply Util/Emacs/README:1]: Yes: in the README :)

File: Makefile (hunk 2 above)
Comment (cpitclaudel) [reply Makefile:18]: It's not used

Comment (cpitclaudel) [reply Makefile:18]: Well, I guess that's not right.  Let me revert this bit.

File: Makefile [hunk 4]
```diff
@@ -15,9 +15,6 @@ ${DIR}/boogie/Binaries/Boogie.exe:
 refman:
 	make -C ${DIR}/docs/DafnyRef
```
Comment (cpitclaudel) [inline Makefile:17]:
```suggestion

refman-release:
	make -C ${DIR}/docs/DafnyRef release
```

File: Makefile [hunk 5]
```diff
@@ -17,7 +17,6 @@ refman:
 
 refman-release:
 	make -C ${DIR}/docs/DafnyRef release
```
Comment (cpitclaudel) [inline Makefile:19]:
```suggestion
	make -C ${DIR}/docs/DafnyRef release

```

File: Makefile (hunk 3 above)
[the remark this answers is not in the document]
Comment (cpitclaudel) [reply Makefile]: I re-added it.  The `package.py` script had support for calling it, but we don't use that anymore.

**Review by MikaelMayer — APPROVED** [review]
